src/worker/exchanges/phemex.ts

```diff
@@ -0,0 +1,93 @@
+import Exchange from '../exchange'
+
+export default class extends Exchange {
+  id = 'PHEMEX'
+
+  protected endpoints = {
+    PRODUCTS: 'https://api.phemex.com/exchange/public/cfg/v2/products'
+  }
+
+  private channels: { [id: string]: string } = {}
+
+  getUrl() {
+    return 'wss://phemex.com/ws'
+  }
+
+  formatProducts(data) {
+    console.log(data);
```

Comment:
this should be where we store the contract size for each pairs
![image](https://user-images.githubusercontent.com/5738336/123286038-d92f9f00-d50d-11eb-8668-6a9180e43172.png)

if format Products return an object, the property of the object are injected in the exchange class,
```
formatProducts(data) {
    const specs = {}
    const products = []

    for (const product of data.data.products) {
      if (product.contractSize) {
        specs[product.symbol] = product.contractSize
      }

      products.push(product.symbol)
    }

    return {
      specs,
      products
    }
  }
```
so then we can use this.specs[* symbol id*] to get the right size of each trade

They seems to have weird price too, scaled price, no idea wtf it is but we could use priceScale / priceRatio from the products endpoint

![image](https://user-images.githubusercontent.com/5738336/123287315-f2851b00-d50e-11eb-99b8-902c03c9a74d.png)

And use it with contractSize

Reply:
Ahh yes thank you for diving deeper into the contract size / price scale.

What I had used was the table values from both contract and spot
![image](https://user-images.githubusercontent.com/2179775/123289683-49bac980-d4d6-11eb-859b-40089023f1ac.png)
![image](https://user-images.githubusercontent.com/2179775/123289744-563f2200-d4d6-11eb-9efc-090f2cf38d71.png)
![image](https://user-images.githubusercontent.com/2179775/123289872-6fe06980-d4d6-11eb-99e9-de8755503665.png)

Did the math work out to get the correct prices?

Reply:
cctx has phemex already, so I'm looking into how they are calculating it https://github.com/ccxt/ccxt/blob/b443e0d94ed81df04d76ab7ad3ba93a25859ba64/js/phemex.js#L824

Reply:
What you did for the price seems to work just fine, I think we don't bother using priceScale and just do the /10000 for perp and /10000 on top of it for spot

Reply:
![image](https://user-images.githubusercontent.com/5738336/123298511-dc7c5800-d518-11eb-8b5b-4b5a4eb527b1.png)
 for size tho they display it in contract, we display in USD

for eth 1 contract is 0.005 eth so when phemex ui report a trade with 1 size
we need to see about 10$ in aggr (1 * 0.005 * 1985 = 9.925)

Reply:
It might be better to use the priceScale just incase they decide to change it in the future

Reply:
Oh for sure, if you figured it out
From all exchanges apis I had to study I never seen anything like this lmao

Reply:
There are three cases:

BTCUSD (perp) --- specific case where the contract currency is in USD
SPOT -- Ez calculation
PERP -- all contract currencies are in ETH/SOL/LTC

I pushed the changes to my master

Reply:
![image](https://user-images.githubusercontent.com/2179775/123305820-2861d980-d4e6-11eb-94a3-7ecd43e9e26e.png)